### PR TITLE
Add automated testing for electron app binaries

### PR DESCRIPTION
**Pull request by ZIMkaRU**
This PR adds automated testing for electron app binaries

---

To be sure that the release build of the Electron app has been successful adds ability to auto-validate that the application works fine on all OSs (Linux, Mac, Win):
- https://www.electronjs.org/docs/latest/tutorial/automated-testing
- https://webdriver.io/docs/desktop-testing/electron
- https://webdriver.io/docs/wdio-electron-service
- https://github.com/webdriverio/webdriverio

---

The flow:
- build release on GitHub Actions
- use unpacked builds for E2E tests
- launch E2E test on Linux, Mac, Win OSs independently
- provide E2E test reports for every OS launch

---

Basic changes:
- adds simple E2E using `WebdriverIO` for verifying binaries
- prevents removal app unpacked folder for E2E test runner
- reworks `electron-builder` config to be able to launch E2E tests
- adds E2E test runner to GitHub actions
- add E2E test report workflow
- skips redundant files in build
- bumps `Electron` up to `27.0.4`
- bump `electron-builder` up to `24.8.1` to have executable name fix
- bumps `actions/checkout` up to v4
- adds `cross-env` for `dev/test` scripts
- adds `local` GitHub Action to turn uninterrupted E2E testing on macOS https://github.com/actions/runner-images/issues/978#issuecomment-652704279
- uses `macos-12` in GitHub Actions
- disables enforce macos app location for E2E test runner

---

Examples of E2E test runners and reports:
- [Build and E2E Test Run](https://github.com/ZIMkaRU/bfx-report-electron/actions/runs/6824031863)
- [E2E Test Report Run](https://github.com/ZIMkaRU/bfx-report-electron/actions/runs/6824171873)
- [Linux report](https://github.com/ZIMkaRU/bfx-report-electron/runs/18559534481)
- [Mac report](https://github.com/ZIMkaRU/bfx-report-electron/runs/18559536796)
- [Win report](https://github.com/ZIMkaRU/bfx-report-electron/runs/18559535680)

Screenshot from 2023-11-10 13-21-17

Screenshot from 2023-11-10 13-21-57

Screenshot from 2023-11-10 13-22-17

Screenshot from 2023-11-08 12-54-28

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,11 +35,18 @@
   },
   "devDependencies": {
     "@mapbox/node-pre-gyp": "1.0.6",
+    "@wdio/cli": "8.22.1",
```
Comment (ezewer) [inline package.json:38]: Can we avoid using @wdio ?

**Review by ezewer — CHANGES_REQUESTED** [review]
small request, rest looks good!

File: package.json (hunk 1 above)
Comment (ZIMkaRU) [reply package.json:38]: resolved, we're moving forward with the `WebdriverIO`

**Review by ezewer — APPROVED** [review]
looks good to me!

**Pull request merged**